### PR TITLE
[UNR-3683] Fix initial server checkout of auth entities

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
https://improbableio.atlassian.net/browse/UNR-3683

#### Release note
Fix for unreleased feature

#### Primary reviewers
@jacquese @samiwh @mironec @sjtsilvester

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp [hunk 1]
```diff
@@ -206,7 +206,10 @@ void InterestFactory::AddServerSelfInterest(Interest& OutInterest, const Worker_
 	// Add a query for components all servers need to read client data
 	Query ClientQuery;
 	ClientQuery.Constraint.EntityIdConstraint = EntityId;
-	ClientQuery.ResultComponentIds = ServerAuthInterestResultType;
+	// Temp fix for invalid initial auth server checkout constraints - UNR-3683
+	// Using full snapshot ensure all components are available on checkout. Remove when root issue is resolved.
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp:210]:
```suggestion
	// Using full snapshot ensures all components are available on checkout. Remove when root issue is resolved.
```
Ultra nit

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp [hunk 2]
```diff
@@ -206,7 +206,10 @@ void InterestFactory::AddServerSelfInterest(Interest& OutInterest, const Worker_
...
+	//ClientQuery.ResultComponentIds = ServerAuthInterestResultType;
```
Comment (jacquese) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp:211]: nit: remove line, ticket should be enough to be able to fix this without leaving commented out code here

Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp:211]:
```suggestion
```

**Comment by m-samiec** [comment]
@sjtsilvester has offer to help validate this fix with his use-case

**Pull request merged**